universal_transfer_operator/src/universal_transfer_operator/data_providers/database/base.py

```diff
@@ -0,0 +1,614 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import pandas as pd
+import sqlalchemy
+
+if TYPE_CHECKING:  # pragma: no cover
+    from sqlalchemy.engine.cursor import CursorResult
+
+import warnings
+
+import attr
+from airflow.hooks.dbapi import DbApiHook
+from pandas.io.sql import SQLDatabase
+from sqlalchemy.sql import ClauseElement
+
+from universal_transfer_operator.constants import (
+    DEFAULT_CHUNK_SIZE,
+    ColumnCapitalization,
+    LoadExistStrategy,
+    Location,
+)
+from universal_transfer_operator.data_providers.base import DataProviders
+from universal_transfer_operator.data_providers.filesystem import resolve_file_path_pattern
+from universal_transfer_operator.data_providers.filesystem.base import FileStream
+from universal_transfer_operator.datasets.base import Dataset
+from universal_transfer_operator.datasets.file.base import File
+from universal_transfer_operator.datasets.table import Metadata, Table
+from universal_transfer_operator.settings import LOAD_TABLE_AUTODETECT_ROWS_COUNT, SCHEMA
+from universal_transfer_operator.universal_transfer_operator import TransferParameters
+from universal_transfer_operator.utils import get_dataset_connection_type
+
+
+class DatabaseDataProvider(DataProviders):
+    """DatabaseProviders represent all the DataProviders interactions with Databases."""
+
+    _create_schema_statement: str = "CREATE SCHEMA IF NOT EXISTS {}"
+    _drop_table_statement: str = "DROP TABLE IF EXISTS {}"
+    _create_table_statement: str = "CREATE TABLE IF NOT EXISTS {} AS {}"
+    # Used to normalize the ndjson when appending fields in nested fields.
+    # Example -
+    #   ndjson - {'a': {'b': 'val'}}
+    #   the col names generated is 'a.b'. char '.' maybe an illegal char in some db's col name.
+    # Contains the illegal char and there replacement, where the value in
+    # illegal_column_name_chars[0] will be replaced by value in illegal_column_name_chars_replacement[0]
+    illegal_column_name_chars: list[str] = []
+    illegal_column_name_chars_replacement: list[str] = []
```

Comment:
When are these values set? Does somebody create the class and then do "provider. illegal_column_name_chars = []"? If so it might be cleaner to turn these into properties.

Reply:
@dimberman They were added in PythonSDK to flatten NDJSON files,  currently we are using them for - 
1. https://github.com/astronomer/astro-sdk/blob/9e94cb59b062060f758f142a82e4060c1a41c45f/python-sdk/src/astro/databases/aws/redshift.py#L75
2. https://github.com/astronomer/astro-sdk/blob/9e94cb59b062060f758f142a82e4060c1a41c45f/python-sdk/src/astro/databases/postgres.py#L30
3. https://github.com/astronomer/astro-sdk/blob/9e94cb59b062060f758f142a82e4060c1a41c45f/python-sdk/src/astro/databases/google/bigquery.py#L87

They are meant to hold data and we don't expect user of the class to change this on the fly.